database/fileindex.go

```diff
@@ -0,0 +1,131 @@
+package database
+
+import (
+	"github.com/activecm/rita/parser/fileparsetypes"
+	"github.com/globalsign/mgo"
+	"github.com/globalsign/mgo/bson"
+	log "github.com/sirupsen/logrus"
+)
+
+type (
+	//ImportedFilesIndex maintains a persistent index over
+	//the files that RITA has imported. This aids in preventing
+	//duplicate imports.
+	ImportedFilesIndex struct {
+		dbHandle            *mgo.Session
+		metaDatabaseName    string
+		indexCollectionName string
+		log                 *log.Logger
+	}
+)
+
+//NewImportedFilesIndex creates a new ImportedFilesIndex
+//and sets up a backing MongoDB collection to maintain the index in.
+//The backing collection will be stored in the MetaDatabase
+//given by metaDatabaseName. The collection will be
+//called indexCollectionName.
+func NewImportedFilesIndex(dbHandle *mgo.Session,
+	metaDatabaseName string, indexCollectionName string,
+	logger *log.Logger) (ImportedFilesIndex, error) {
+
+	//ensure the index collection is indexed by file hash + database
+	fileIndex := mgo.Index{
+		Key:        []string{"hash", "database"},
+		Unique:     true,
+		DropDups:   true,
+		Background: true, //this is a hold over. I don't think we want this. (LL)
+		Name:       "hashindex",
+	}
+
+	//creating an index is idempotent so it won't error if the index already
+	//exists. Also collections are created implicitly, so we can create
+	//the collection and ensure the index exists in one fell swoop.
+	err := dbHandle.DB(metaDatabaseName).C(indexCollectionName).EnsureIndex(fileIndex)
+	if err != nil {
+		logger.WithFields(log.Fields{
+			"metadatabase":     metaDatabaseName,
+			"index_collection": indexCollectionName,
+			"error":            err.Error(),
+		}).Error("Failed to create index for the imported file index")
+		return ImportedFilesIndex{}, err
+	}
+
+	index := ImportedFilesIndex{
+		dbHandle:            dbHandle.Copy(),
+		metaDatabaseName:    metaDatabaseName,
+		indexCollectionName: indexCollectionName,
+		log:                 logger,
+	}
+
+	return index, nil
+}
+
+func (i *ImportedFilesIndex) openIndexCollection() *mgo.Collection {
```

Comment:
Same single letter variable issue as above, not needed now but we should work away from these